docs/design/windows-images.md

```diff
@@ -0,0 +1,164 @@
+
+All of this needs to be cleaned up so that we can get avoid deployment time penalties due to downloads and keep all changes in dependencies in source code so that they can be thoroughly tested and reproduced. 
+
+## Use Cases
+
+In general, most code build for AKS-Engine deployments should go through a VHD-based workflow to match what will be used in production and avoid having extra implementations for the same scenario. Ones that are intentionally overlapping should be called out.
+
+### Testing new Windows versions - VHD
+
+Testing new Windows versions is particularly difficult for a few reasons:
+
+- Windows Server Insider builds are published frequently, but not all are available on Azure. Starting from an ISO+Packer works around that limitation.
+- The base containers are different -  `nanoserver-insider` vs `nanoserver`, and are incompatible with the released versions. This means that images Kubernetes depends on such as `pause` need to be built and used as well. You can't build these on an older OS, but they can be done within the VM that's running as part of the VHD build process.
+  - As an aside, matching test containers are required too. If these aren't built from Kubernetes official builds, they could be built and pushed to a private registry using the same VM. Leaving them on the VHD will actually improve test time by avoiding pulls later.
+
+With the VHD-based workflow, adding (or removing) support for a new version could be reduced to a few parameters: `WindowsIsoUrl`, `NanoServerImage`. If these are in a pull request, automation can run the full vhd build followed by AKS-Engine E2E test pass.
+
+If a private build needed to be tested, `WindowsIsoUrl` could be a path requiring an Azure login, and the resulting image could be kept in the private store for that subscription.
+
+
+### Testing new Windows patches - VHD
+
+This is similar, but simpler than the new Windows version case above. It could use a new `WindowsIsoUrl`, `WindowsPatchList`, or `WindowsPrivatePatchUrl`. There are no changes needed to the container images. 
+
+Teams needing to create and test prerelease patches will need to use a private branch and Azure DevOps Pipeline to avoid unintentional security disclosures. This should be kept in-sync with the public repo but use a different Azure subscription and accounts.
+
+### AKS-Users testing new Windows patches
+
+During the VHD build, `WindowsPatchList` and `WindowsPrivatePatchList` control what cumulative updates and/or patches are installed. However, there may be cases where a private patch has been given to a customer who cannot build a new VHD. The existing [windows-patches](C:\Users\patri\Source\aks-engine\extensions\windows-patches) extension should still work to do this.
+

```

Comment:
Is this AKS-Engine users? I don't see how AKS users can test this without falling back to AKS-Engine.

Reply:
sorry yes that should be aks-engine